apps/chat/src/components/Chat/Chat.tsx

```diff
@@ -468,8 +468,12 @@ export const ChatView = memo(() => {
   );
 
   const handleDeleteMessage = useCallback(
-    (index: number) => {
-      dispatch(ConversationsActions.deleteMessage({ index }));
+    (index: number, conv: Conversation) => {
+      let finalIndex = index;
+      if (conv.messages.at(0)?.role === Role.System) {
+        finalIndex += 1;
+      }
+      dispatch(ConversationsActions.deleteMessage({ index: finalIndex }));
```

Comment:
```suggestion
      dispatch(ConversationsActions.deleteMessage({ index: conv.messages.at(0)?.role === Role.System ? index + 1 : index }));
```

Maybe `finalIndex` is redundant?

Reply:
Actually I don't want to have this logic just in ternary to have it more explicitly